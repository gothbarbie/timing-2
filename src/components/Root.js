import React from 'react'
import PropTypes from 'prop-types'
import { Provider } from 'react-redux'

import defaultStore from '../store/configureStore'

const Root = ({ store, children }) => (
  <Provider store={store ? store : defaultStore}>{children}</Provider>
)

Root.propTypes = {
  store: PropTypes.shape({}),
  children: PropTypes.node,
}

export default Root
