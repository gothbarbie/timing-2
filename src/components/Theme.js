import React from 'react'
import PropTypes from 'prop-types'
import { ThemeProvider } from 'styled-components'

import theme from './themes/default'

const Theme = ({ children }) => (
  <ThemeProvider theme={theme}>{children}</ThemeProvider>
)

Theme.propTypes = {
  children: PropTypes.node,
}

export default Theme
