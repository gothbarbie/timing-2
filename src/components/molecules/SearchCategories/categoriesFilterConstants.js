export const SET_CATEGORIES_FILTER = 'SET_CATEGORIES_FILTER'
