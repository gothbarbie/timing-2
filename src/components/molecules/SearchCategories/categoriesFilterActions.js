import { SET_CATEGORIES_FILTER } from './categoriesFilterConstants'

export const setCategoriesFilter = filter => ({
  type: SET_CATEGORIES_FILTER,
  payload: filter,
})
