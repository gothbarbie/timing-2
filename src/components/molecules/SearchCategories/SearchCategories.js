import React from 'react'
import PropTypes from 'prop-types'
import styled from 'styled-components'
import { connect } from 'react-redux'

import { setCategoriesFilter } from './categoriesFilterActions'
import { getFilteredCategories } from './categoriesSelectors'

import Icon from '../../atoms/Icon'
import Button from '../../atoms/Button'

import LabeledInput from '../LabeledInput'

import Category from '../Category'

const FormSection = styled.section`
  display: grid;
  grid-template-columns: 4fr 1fr;
  grid-gap: 1rem;
  margin: 0 0 3rem 0;
  align-items: flex-end;
`
const SearchResults = styled.section`
  display: flex;
  margin-bottom: 2rem;
`
const NoResults = styled.div`
  display: flex;
  justify-content: center;
  align-items: center;
  width: 100%;
  margin: 3rem 0;
  font-size: 1.4rem;
  font-weight: 500;
  color: ${({ theme }) => theme.palette.grayScale[0]};
`

class SearchCategories extends React.PureComponent {
  handleSetCategoriesFilter = event => {
    console.log('handleSetCategoriesFilter', event.target.value)
    this.props.setCategoriesFilter(event.target.value)
  }

  render() {
    console.log('categories', this.props.categories)
    const { categories } = this.props
    return (
      <>
        <FormSection>
          <LabeledInput
            label="Search"
            name="search-category"
            onChange={this.handleSetCategoriesFilter}
            type="search"
          />
          <Button hasIcon primary>
            <Icon icon={['fas', 'search']} />
          </Button>
        </FormSection>

        <SearchResults>
          {!categories.length && <NoResults>Search to get results</NoResults>}
          {categories.map(category => (
            <Category
              id={category[0]}
              key={category[0]}
              name={category[1].name}
            />
          ))}
        </SearchResults>
      </>
    )
  }
}

SearchCategories.propTypes = {
  categories: PropTypes.arrayOf(
    PropTypes.shape({
      key: PropTypes.string.isRequired,
      name: PropTypes.string.isRequired,
    }).isRequired
  ),
  setCategoriesFilter: PropTypes.func.isRequired,
}

const mapStateToProps = state => ({
  categories: getFilteredCategories(state),
})

const mapDispatchToProps = {
  setCategoriesFilter,
}

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(SearchCategories)
