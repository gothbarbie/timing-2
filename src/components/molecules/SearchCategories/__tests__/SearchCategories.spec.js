import React from 'react'
import { render } from 'enzyme'
import { library } from '@fortawesome/fontawesome-svg-core'
import { faSearch } from '@fortawesome/free-solid-svg-icons'

import Root from '../../../Root'
import Theme from '../../../Theme'
import SearchCategories from '../SearchCategories'

library.add(faSearch)

describe('components/molecules/SearchCategories', () => {
  it('renders', () => {
    const props = {
      categories: [{ key: '123', name: 'name' }],
    }
    const component = render(
      <Root>
        <Theme>
          <SearchCategories {...props} />
        </Theme>
      </Root>
    )

    expect(component).toMatchSnapshot()
  })
})
