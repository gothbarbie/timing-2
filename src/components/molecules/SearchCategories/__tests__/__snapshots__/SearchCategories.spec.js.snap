// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/molecules/SearchCategories renders 1`] = `
Array [
  <section
    class="FormSection-kjv4fi-0 dEHpOz"
  >
    <div
      class="LabeledInputStyle-haayko-0 hPxfKk"
    >
      <label
        class="LabelStyled-haayko-1 iNeUPg"
        for="search-category"
      >
        Search
      </label>
      <input
        class="InputStyle-sc-1jmrrj1-0 jTNhfr"
        name="search-category"
        type="search"
      />
    </div>
    <button
      class="Primary-sc-12a0ki3-2 eysKsr"
    >
      <svg
        aria-hidden="true"
        class="svg-inline--fa fa-search fa-w-16 IconStyle-sc-1dp848p-0 iFEbEV"
        data-icon="search"
        data-prefix="fas"
        role="img"
        viewBox="0 0 512 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M505 442.7L405.3 343c-4.5-4.5-10.6-7-17-7H372c27.6-35.3 44-79.7 44-128C416 93.1 322.9 0 208 0S0 93.1 0 208s93.1 208 208 208c48.3 0 92.7-16.4 128-44v16.3c0 6.4 2.5 12.5 7 17l99.7 99.7c9.4 9.4 24.6 9.4 33.9 0l28.3-28.3c9.4-9.4 9.4-24.6.1-34zM208 336c-70.7 0-128-57.2-128-128 0-70.7 57.2-128 128-128 70.7 0 128 57.2 128 128 0 70.7-57.2 128-128 128z"
          fill="currentColor"
        />
      </svg>
    </button>
  </section>,
  <section
    class="SearchResults-kjv4fi-1 jsvAnJ"
  >
    <div
      class="NoResults-kjv4fi-2 GhGmn"
    >
      Search to get results
    </div>
  </section>,
]
`;
